hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -155,14 +164,155 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
+    }
+
+    try {
+      clients.run(client -> {
+        client.dropDatabase(namespace.level(0),
+            false /* deleteData */,
+            false /* ignoreUnknownDb */,
+            true /* cascade */);
+        return null;
+      });
+
+      return true;
+
+    } catch (NoSuchObjectException e) {
+      throw new NoSuchNamespaceException(e, "Namespace '%s' does not exist!", namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop namespace " + namespace + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to drop dropDatabase(name)" + namespace + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public boolean alterNamespace(Namespace namespace, ImmutableMap<String, String> meta) {
+    if (meta.containsKey("location")) {
+      throw new UnsupportedOperationException("Does not support change 'location' in HiveCatalog: " + namespace);
+    }
+
+    if (meta.containsKey("name")) {
+      throw new UnsupportedOperationException("Does not support change 'name' in HiveCatalog: " + namespace);
+    }
+
+    Map<String, String> newMeta = Maps.newHashMap();
+    newMeta.putAll(loadNamespaceMetadata(namespace));
+    newMeta.putAll(meta);
+
+    try {
+      clients.run(client -> {
+        client.alterDatabase(namespace.toString(), convertToDatabase(namespace, newMeta));
+        return null;
+      });
+
+      return true;
+
+    } catch (NoSuchObjectException | UnknownDBException e) {
+      throw new NoSuchNamespaceException(e, "Namespace does not exist: %s", namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException(
+          "Failed to list namespace under namespace: " + namespace + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to getDatabase(name)" + namespace + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
+    if (namespace.levels().length != 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    try {
+      Database database = clients.run(client -> client.getDatabase(namespace.toString()));
+      return convertToMetadata(database);
+
+    } catch (NoSuchObjectException | UnknownDBException e) {
+      throw new NoSuchNamespaceException(e, "Namespace does not exist: %s", namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list namespace under namespace: " + namespace + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to getDatabase(name)" + namespace + " in Hive MataStore", e);
```

Comment:
Nit: missing space between `getDatabase(name)` and `namespace` in the error message. You may find it easier to avoid these problems by using `String.format`.